src/sentry/api/event_search.py

```diff
@@ -558,6 +587,11 @@ def get_snuba_query_args(query=None, params=None):
             raise InvalidSearchQuery(
                 u'Parse error: %r (column %d)' % (e.expr.name, e.column())
             )
+        except IncompleteParseError as e:
+            raise InvalidSearchQuery(
+                'Parse error: Search did not parse completely. This is commonly caused by unmatched-parenthesis. '
+                + six.text_type(e)
```

Comment:
Are we able to give the user a more specific error on where their query failed to be parsed? Like a column or token?

Reply:
I’ll add it the whole message to the test for visibility. But it tells you where the unmatched string began. It’s not that helpful most likely because it’s the outermost  open paren of the unmatched group of parentheses. But it is there. I’ll add it to the test so you can see the whole thing.